app/views/rooms/_form.html.erb

```diff
@@ -3,22 +3,29 @@
     <%= render "text_field", attribute: :name, form: room_form %>
     <%= render "text_field", attribute: :slug, form: room_form %>
     <%= render "text_area", attribute: :description, form: room_form, label_hint: 'Add brief summary of your section for search engines to display as snippets in results' %>
+    <h3>Privacy and Security</h3>
+    <%= render "radio_group", attribute: :access_level, options: Room::access_levels.values, form: room_form %>
+    <footer>
+      <%= room_form.submit %>
+    </footer>
+  <% end %>
+  <%=  %>
+  <%= form_with url:polymorphic_path(room.location(child: :images))  do |room_images_form| %>
```

Comment:
The more resource-oriented/railsy way to do this would be `form_with(model: (room.hero_image || room.build_hero_image).location) do |hero_image_form|`.

This would also get rid of the need to pass in a `model` to the `file_field` helper.